cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceTaskContext.java

```diff
@@ -296,13 +297,22 @@ public MessageFetcher getMessageFetcher() {
     };
   }
 
-  private static Map<String, String> createMetricsTags(@Nullable String taskId,
+  private static Map<String, String> createMetricsTags(ProgramOptions programOptions,
+                                                       @Nullable String taskId,
                                                        @Nullable MapReduceMetrics.TaskType type,
                                                        @Nullable WorkflowProgramInfo workflowProgramInfo) {
     Map<String, String> tags = Maps.newHashMap();
-    if (type != null && taskId != null) {
+    if (type != null) {
       tags.put(Constants.Metrics.Tag.MR_TASK_TYPE, type.getId());
-      tags.put(Constants.Metrics.Tag.INSTANCE_ID, taskId);
+
+      if (taskId != null) {
+        String taskMetricsPreference =
```

Comment:
So this is not disabling emission of metric but rather just not to have task id in the context?? I don't get it.

Reply:
yes, one of the requirement is `have a way to disable (by config/preference) the task-level metrics (instead only record by task type) to avoid the many writes to the metrics table` if we do `task.*.metrics.enabled=false` we wont get any metrics from the task. rather we want those metrics but not at instance-id level.

Reply:
I see. Then better call the option `system.metrics.context.task.included`. If you have `enable` in the name, it sounds like it is enabling/disabling the metrics.